classes/common/class-checkout-modifications.php

```diff
@@ -217,4 +236,46 @@ public static function attach_order_note( $order_id ) {
 			__( 'Customer requested delivery time:', 'bring-fraktguiden-for-woocommerce' ) . " $time_slot"
 		);
 	}
+
+	/**
+	 * Bag on door consent
+	 */
+	public static function bag_on_door_consent() {
+		$current_shipping_method = WC()->session->get( 'chosen_shipping_methods' );
+		
+		if ( ! empty ( $current_shipping_method ) && $current_shipping_method[0] === 'bring_fraktguiden:3584' ) {
+			$name = __FUNCTION__;
+	
+			echo "<div id='$name'>";
+			woocommerce_form_field($name, array(
+				'type'      => 'checkbox',
+				'class'     => array('input-checkbox'),
+				'label'     => __( 'I agree to have my package delivered with bag on door', 'bring-fraktguiden-for-woocommerce' ),
+				'required'	=> true
+			), WC()->checkout->get_value($name));
+			echo '</div>';
+		}
+	}
+
+	/**
+	 * Save user selected bag on door value to order meta
+	 */
+	public static function bag_on_door_order_meta( $order_id ) {
+		$name = "bag_on_door_consent";
```

Comment:
Ambiguous naming. Also, try not to assign machine keys or slugs to variables. Instead use the full string where it's needed.